### PR TITLE
feat(fabric-connector): add private data support

### LINKED ISSUE #857: feat(connector-fabric): private data collections support
**Is your feature request related to a problem? Please describe.**

Same thing that is being tutored here should work through the Fabric connector plugin of Cactus as well.
https://developer.ibm.com/recipes/tutorials/implement-private-data-collections-with-fabric-a-practical-example/

**Describe the solution you'd like**

The ability to specify the transient data in the run transaction request when invoking a Fabric chain code contract. 

**Describe alternatives you've considered**

Nothing, I'm pretty confident that we need this as a feature.

**Additional context**

Relevant issue from the carbon accounting tool:
https://github.com/hyperledger-labs/blockchain-carbon-accounting/issues/6

cc: @takeutak @sfuji822 @hartm @jonathan-m-hamilton @AzaharaC @jordigiam @kikoncuo @jagpreetsinghsasan @arnab-roy

### PR DESCRIPTION
**Pull request by RafaelAPB**
Private Data Collections (PDC) to provide storage and retrieval of private data on the blockchain network for authorized peers of organizations. 

Private data changes Fabric's flows in mainly two ways:
1) smart contract deployment has as input a private data collection, "allowing a defined subset of organizations on a channel the ability to endorse, commit, or query private data without having to create a separate channel."
2) some transaction proposals require a transient field
3) private data is shared via gossip (instead of using the orderer)

According to Fabric's documentation, "Note that because gossip distributes the private data peer-to-peer across authorized organizations, it is required to set up anchor peers on the channel, and configure CORE_PEER_GOSSIP_EXTERNALENDPOINT on each peer, in order to bootstrap cross-organization communication".

This PR adds support to use private data capabilities on the Fabric Connector, by leveraging the transient data field. Also supports setting endorsing parties.

A test case was added (deploy chaincode and private data collections; create an asset on a private data collection; query assets).

closes #857

### CODE REVIEW

**Review by takeutak — APPROVED** [review]

File: packages/cactus-plugin-ledger-connector-fabric/src/test/typescript/integration/fabric-v2-2-x/deploy-cc-from-golang-source-private-data.test.ts [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+
+  // CreateAsset(id string, color string, size int, owner string, appraisedValue int)
+  const createRes = await apiClient.runTransactionV1({
+    transientData: rawTmpData,
+    contractName,
+    channelName,
+    //objectType, assetID, color, size, appraisedvalue
+    params: [],
+    methodName: "CreateAsset",
+    invocationType: FabricContractInvocationType.SENDPRIVATE,
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-fabric/src/test/typescript/integration/fabric-v2-2-x/deploy-cc-from-golang-source-private-data.test.ts:316]: @RafaelAPB Is there something in the output of the call that we could further use to validate that the whole thing works or not? Or does it not crashing guarantees that everything related to the private data (including the initial collection config json) was working as expected? Sorry if this is a beginner question I still haven't had time to read up on how exactly Fabric does this so I'm just checking.

File: packages/cactus-plugin-ledger-connector-fabric/src/test/typescript/integration/fabric-v2-2-x/deploy-cc-from-golang-source-private-data.test.ts [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+  );
+  const privateDataCollections = {
+    body: privateDataCollectionsBuf.toString("base64"),
+    filename: privateDataCollectionName,
+  };
+
+  const res = await apiClient.deployContractV1({
+    channelId,
+    ccVersion: "1.0.0",
+    sourceFiles: [
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-fabric/src/test/typescript/integration/fabric-v2-2-x/deploy-cc-from-golang-source-private-data.test.ts:228]: @RafaelAPB If these are the only files needed then I'd remove the other files from the fixtures directory to reduce the size of the PR if possible (trying to keep the git repo lean where possible).
If you have plans to add test coverage for the additional files (such as the java version of the example chain code) then I withdraw my suggestion.

**Review by petermetz — CHANGES_REQUESTED** [review]

File: packages/cactus-plugin-ledger-connector-fabric/src/test/typescript/integration/fabric-v2-2-x/deploy-cc-from-golang-source-private-data.test.ts (hunk 1 above)
Comment (RafaelAPB) [reply packages/cactus-plugin-ledger-connector-fabric/src/test/typescript/integration/fabric-v2-2-x/deploy-cc-from-golang-source-private-data.test.ts:316]: Good point - I believe the tx receipt is the same. The test cases show that this feature is indeed working (but can be expanded)

File: packages/cactus-plugin-ledger-connector-fabric/src/test/typescript/integration/fabric-v2-2-x/deploy-cc-from-golang-source-private-data.test.ts (hunk 2 above)
Comment (RafaelAPB) [reply packages/cactus-plugin-ledger-connector-fabric/src/test/typescript/integration/fabric-v2-2-x/deploy-cc-from-golang-source-private-data.test.ts:228]: Ups - that was a mistake. Those should not be there. Thanks!

**Comment by RafaelAPB** [comment]
Should be fine and good to go, @petermetz !

**Comment by petermetz** [comment]
@RafaelAPB I see these being used by the test:

...t/typescript/fixtures/go/asset-transfer-private-data/chaincode-go/collections_config.json 

  25  ...or-fabric/src/test/typescript/fixtures/go/asset-transfer-private-data/chaincode-go/go.mod 
Viewed
  234  ...or-fabric/src/test/typescript/fixtures/go/asset-transfer-private-data/chaincode-go/go.sum 
Viewed
  23  ...r-fabric/src/test/typescript/fixtures/go/asset-transfer-private-data/chaincode-go/main.go 
Viewed


But then I also see these test and mock files written in go:

  185  ...ript/fixtures/go/asset-transfer-private-data/chaincode-go/chaincode/asset_queries_test.go 
Viewed
  767  ...pescript/fixtures/go/asset-transfer-private-data/chaincode-go/chaincode/asset_transfer.go 
Viewed
  444  ...ipt/fixtures/go/asset-transfer-private-data/chaincode-go/chaincode/asset_transfer_test.go 
Viewed
  2,878  ...ipt/fixtures/go/asset-transfer-private-data/chaincode-go/chaincode/mocks/chaincodestub.go 
Viewed
  399  ...pt/fixtures/go/asset-transfer-private-data/chaincode-go/chaincode/mocks/clientIdentity.go 
Viewed
  232  ...ixtures/go/asset-transfer-private-data/chaincode-go/chaincode/mocks/statequeryiterator.go 
Viewed
  164  ...cript/fixtures/go/asset-transfer-private-data/chaincode-go/chaincode/mocks/transaction.go 


I wouldn't mind leaving those in if these tests are actually executable and passing, but otherwise it's probably wiser to delete them as well to reduce the cognitive load of the fixture.
This is just a nit pick, apologies in advance.
So the question is, can you confirm that these tests work/pass on your machine at least? (if not that's fine, but then I'd ask to drop these as well because the test case doesn't use them)

**Comment by RafaelAPB** [comment]
Done, @petermetz , thanks for the feedback :)

File: packages/cactus-plugin-ledger-connector-fabric/src/test/typescript/integration/fabric-v2-2-x/deploy-cc-from-golang-source-private-data.test.ts (hunk 1 above)
Comment (petermetz) [reply packages/cactus-plugin-ledger-connector-fabric/src/test/typescript/integration/fabric-v2-2-x/deploy-cc-from-golang-source-private-data.test.ts:316]: @RafaelAPB Understood, thank you for the explanation!

**Review by petermetz — APPROVED** [review]

**Comment by petermetz** [comment]
@RafaelAPB No worries, thank you for the updates to the PR!
FYI: This failure in the CI is detected that something in your changes made it so that the build or the test have files changing and/or getting added to version control which is a problem because either these changes should be under version control so that the build is stable OR the files that are being added by the build should be ignored by git (so that the final outcome is that the build does not put anything in the git index).
If you tell me what files are being added/changed I can also provide more details/guidance on how to deal with them but this is the generic explanation I could give without that info so far.

**Comment by RafaelAPB** [comment]
I'm not sure @petermetz - looks like the ci complains about a diff on packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/generated/openapi/typescript-axios/api.ts, which is under version control.

**Comment by petermetz** [comment]
> a diff on packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/generated/openapi/typescript-axios/api.ts,

@RafaelAPB Yeah, I just noticed that you changed the openapi.json file so the code generator picks up those changes and puts them in .../api.ts. and that's what you need to commit as part of your PR, so just stage those code generator made changes, commit them (squash or amend as required so that the PR is still just a single commit) and then it should start passing.

**Pull request merged**